kbase-extension/static/kbase/js/util/kbaseApiUtil.js

```diff
@@ -62,28 +67,80 @@ define([
      * Checks that a given object reference is valid. Returns true if the reference
      * is of the form xx/yy or xx/yy/zz or xx/yy/zz;aa/bb/cc, or combinations.
      * Basically, if it's a reference or reference path, this returns true.
+     *
+     * Note that if there's a reference path (xx/yy/zz;aa/bb/cc), these MUST include
+     * versions. I.e. xx/yy;aa/bb/cc will fail.
+     * @param {string} ref - a workspace reference string
      */
     function checkObjectRef (ref) {
         // return true if this.options.objRef = a reference or reference path
         // return false otherwise
         if (!ref) {
             return false;
         }
-        var refRegex = /^\S+\/\S+(\/\d+)?$/;
+
+        const refRegex = /^\w+\/\w+(\/\d+)?$/;
+        if (ref.indexOf(';') === -1) { // if it's not a path, just do the basic regex
+            if (refRegex.exec(ref)) {
+                return true;
+            }
+            return false;
+        }
+
+        const versionRefRegex = /^\w+\/\w+\/\d+?$/;
         var refList = ref.split(';');
         var validRef = true;
         refList.forEach(function(r) {
-            if (!refRegex.exec(r)) {
+            if (!versionRefRegex.exec(r)) {
                 validRef = false;
             }
         });
         return validRef;
     }
 
+    /**
+     * This breaks down a workspace type name into its component parts. This returns an object with
+     * structure with these properties:
+     * - module - the module name
+     * - type - the type name
+     * - version - the complete version
+     * - majorVersion - the major part of the version
+     * - minorVersion - the minor part of the version
+     * For example, "KBaseNarrative.Narrative-1.0" would break down to:
+     * {
+     *   module: 'KBaseNarrative',
+     *   type: 'Narrative',
+     *   version: '1.0',
+     *   majorVersion: '1',
+     *   minorVersion: '0'
+     * }
+     * If the string is not a valid workspace type string, this returns null;
+     * @param {string} typeString a workspace type string of format:
+     *      ModuleName.TypeName-MajorVersion.MinorVersion
+     */
+    function parseWorkspaceType (typeString) {
```

Comment:
This will be handy -- I can already think of one place to use this (the icon module), and I'm sure there are numerous other places throughout the codebase.